### PR TITLE
Initial resultstoreui setup

### PR DESCRIPTION
**Pull request by craigericlewis**
Implement resultstoreui CLI functionality to get and create invocations as well as perform single uploads to invocations.

### CODE REVIEW

File: resultstoreui.py [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+
+    Args:
+        target (str): The host and port of the service
+    
+    Returns:
+        A gRPC channel
+    """
+    credentials, _ = auth.default(
+        scopes=("https://www.googleapis.com/auth/cloud-source-tools",
+                "https://www.googleapis.com/auth/cloud-platform"))
```
Comment (J-Pai) [inline resultstoreui.py:79]: Make these scopes a global constant.

Combine with bigstore scopes below in a common credentials file.

**Comment by craigericlewis** [comment]
> 1. Make a separate PR for generated python files and merge that in first.
Opened a new pr with just the generate python files here https://github.com/google/resultstoreui/pull/2

**Comment by craigericlewis** [comment]
Made the request changes from @J-Pai.
1. Moved out resultstore api stuff into a separate client
2. Moved out bigstore uploader function into a separate client
3. Separate credential class
Other minor updates

**Comment by zhm12359** [comment]
I think you should use git commit --amend so the commit logs aren't contaminated. For now you should squash your commits

**Comment by zhm12359** [comment]
Also can you rebase? So that this PR won't show the python generated files.

File: resultstore_client.py [hunk 2]
```diff
@@ -0,0 +1,438 @@
...
+        else:
+            build_action = action_pb2.BuildAction()
+            action.build_action.CopyFrom(build_action)
+        self.create_action(action)
+        self.file_upload_helper(target_id, action.id.action_id, config_id)
+        self.finalize_configured_target(self.flags.invocation_id, target_id,
+                                        config_id)
+        self.finalize_target(self.flags.invocation_id, target_id)
+
+    def file_upload_helper(self, target_id, action_id, config_id):
```
Comment (J-Pai) [inline resultstore_client.py:400]: Is this meant to be publicly accessible? If not, note that it is private.

File: resultstoreui.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
+from absl import (flags, app)
+from resultstore_client import ResultStoreClient
+from credentials import Credentials
+from resultstoreapi.cloud.devtools.resultstore_v2.proto import common_pb2
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string('command', '', 'Command to execute')
```
Comment (zhm12359) [inline resultstoreui.py:7]: list out possible values.

**Pull request merged**